src/auth/OidcBroker.ts

```diff
@@ -0,0 +1,106 @@
+    token_endpoint: `${getConfig().orchestrationUrlRoot}/oauth2/token`,
+  };
+  return {
+    authority: `${getConfig().orchestrationUrlRoot}/oauth2/authorize`,
+    // The clientId from oidcStore.config is not undefined, setup in initializeClientId on appLoad
+    client_id: oidcStore.get().config.clientId!,
+    popup_redirect_uri: `${window.origin}/redirect-from-oauth`,
+    silent_redirect_uri: `${window.origin}/redirect-from-oauth-silent`,
+    metadata,
+    prompt: 'consent login',
+    scope: 'openid email profile',
+    stateStore: new WebStorageStateStore({ store: getLocalStorage() }),
+    userStore: new WebStorageStateStore({ store: getLocalStorage() }),
+    automaticSilentRenew: true,
+    // Leo's setCookie interval is currently 5 min, set refresh auth then 5 min 30 seconds to guarantee that setCookie's token won't expire between 2 setCookie api calls
+    accessTokenExpiringNotificationTimeInSeconds: 330,
+    includeIdTokenInSilentRenew: true,
+    extraQueryParams: { access_type: 'offline' },
+    redirect_uri: '', // this field is not being used currently, but is expected from UserManager
+  };
+};
+
+// This is the first thing that happens on app load.
+export const initializeClientId = _.memoize(async () => {
+  const oidcConfig: OidcConfig = await Ajax().OAuth2.getConfiguration();
+  oidcStore.update((state) => ({ ...state, config: oidcConfig }));
+});
+
+// Instantiate a UserManager directly to populate the logged-in user at app initialization time.
+// All other auth usage should use the AuthContext from oidcStore.
+export const getCurrentOidcUser = async (): Promise<OidcUser | null> => {
+  const userManager: UserManager = new UserManager(getOidcConfig());
+  return userManager.getUser();
+};
+
+export const getAuthInstance = (): AuthContextProps => {
+  const authContext: AuthContextProps | undefined = oidcStore.get().authContext;
+  if (authContext === undefined) {
+    console.error('getAuthInstance must not be called before authContext is initialized.');
+    throw new Error('Error initializing authentication.');
+  }
+  return authContext;
+};
+
+export const getSignInArgs = (includeBillingScope: boolean): ExtraSigninRequestArgs => {
+  return Utils.cond(
+    [!includeBillingScope, (): ExtraSigninRequestArgs => ({})],
+    // For B2C use a dedicated policy endpoint configured for the GCP cloud-billing scope.
+    (): ExtraSigninRequestArgs => ({
+      extraQueryParams: { access_type: 'offline', p: getConfig().b2cBillingPolicy },
+      extraTokenParams: { p: getConfig().b2cBillingPolicy },
+    })
+  );
+};
+export const oidcSignIn = async (includeBillingScope = false, popUp = false): Promise<OidcUser | null> => {
+  const args: ExtraSigninRequestArgs = getSignInArgs(includeBillingScope);
+  const authInstance: AuthContextProps = getAuthInstance();
+
+  return popUp
+    ? // returns Promise<OidcUser | null>, attempts to use the refresh token to get a new authToken
```

Comment:
That's interesting, that seems to imply that whenever you `signinPopup` that gives you a new refresh token since the null response is when the refresh token is expired. I will take a look into that.